### PR TITLE
bugfix: utapi reindex needs configuration to overwrite defaults

### PR DESCRIPTION
**Pull request by JianqinWang**
current config missing password

### CODE REVIEW

File: lib/utapi/utapiReindex.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 const UtapiReindex = require('utapi').UtapiReindex;
 const { config } = require('../Config');
 
-const reindex = new UtapiReindex(config.utapi && config.utapi.reindex);
+const reindexConfig = config.utapi.reindex ? config.utapi.reindex : config.utapi
```
Comment (philipyoo) [inline lib/utapi/utapiReindex.js]: Is config.utapi always defined?

**Review by ssalaues — APPROVED** [review]

File: lib/utapi/utapiReindex.js (hunk 1 above)
Comment (JianqinWang) [reply lib/utapi/utapiReindex.js]: yes - if for some reason it isn't, the UtapiReindex has default values to use

**Comment by JianqinWang** [comment]
waiting to test this and https://github.com/scality/utapi/pull/287 with artifact from https://github.com/scality/Federation/tree/bugfix/S3C-2019-testing-utapi before merging

**Review by philipyoo — APPROVED** [review]

**Comment by JianqinWang** [comment]
@bert-e reset

**Comment by JianqinWang** [comment]
@bert-e reset

File: lib/utapi/utapiReindex.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 const UtapiReindex = require('utapi').UtapiReindex;
 const { config } = require('../Config');
 
-const reindex = new UtapiReindex(config.utapi && config.utapi.reindex);
+const reindexConfig = config.utapi.reindex ?
```
Comment (bennettbuchanan) [inline lib/utapi/utapiReindex.js]: if `config.utapi` is undefined `config.utapi.reindex` will throw an error here.

**Review by bennettbuchanan — CHANGES_REQUESTED** [review]

**Review by bennettbuchanan — APPROVED** [review]

**Comment by JianqinWang** [comment]
@bert-e reset

**Review by jonathan-gramain — APPROVED** [review]

**Comment by jonathan-gramain** [comment]
@bert-e bypass_leader_approval

**Comment by jonathan-gramain** [comment]
@bert-e bypass_author_approval

**Comment by jonathan-gramain** [comment]
After discussion with @flavien-scality, decided to bypass leader approval so that an RC3 can be built today and testing can start.

**Pull request merged**